README.md

```diff
@@ -1,116 +1,130 @@
+
+ *  [adoptopenjdk/openjdk8-openj9:x86_64-ubuntu-jdk8u162-b12_openj9-0.8.0](https://hub.docker.com/r/adoptopenjdk/openjdk8-openj9/)
+
+## Creating a Kotlin Action
+
+The Runtime for Kotlin supports two APIs for creating actions:  
+1. JSON based.  
+The action receives parameters as a JSON object and returns a JSON object, using the [Google GSON library](https://github.com/google/gson) library.  
+2. Data Class based.  
+The action receives parameters as a user defined data Class, and returns a user defined data Class.  
+
+### "Hello World" using the JSON API
+The following shows how to build a "HelloWorld" action using the JSON API:
+
+1. Create a file called `main.kt` containing:
+
+	```kotlin
+	import com.google.gson.JsonObject
+	
+	fun main(args: JsonObject) : JsonObject {
+	    val name = args.getAsJsonPrimitive("name").getAsString();
+	    val hello = JsonObject();
+	    hello.addProperty("greeting", "Hello " + name + "!");
+	    return hello
+	}
+	```
+2. Compile the action into a JAR file:
+
+	```sh
+	kotlinc -classpath ./gson-2.6.2.jar main.kt -d myAction.jar
+	```
+	
+This provides the action contained in myAction.jar, ready to be deployed and run.
+
+### "Hello World" using the Data Class API
+The following shows how to build a "HelloWorld" action using the Data Class API:
+
+1. Create a file called `main.kt` containing:
+
+	```kotlin
+	data class User (
+	    val name: String
+	)
+	
+	data class Hello (
+	    val greeting: String
+	)
+	
+	fun main(user: User) : Hello {
+	    val hello = Hello("Hello " + user.name + "!")
+	    return hello
+	}
+	```
+	
+2. Compile the action into a JAR file:
+
+	```sh
+	kotlinc main.kt -d myAction.jar
+	```
+This provides the action contained in myAction.jar, ready to be deployed and run.
+
+### Deploying the Kotlin Action
+The Kotlin action can be deployed for use as a Docker action using the following, works on any deployment of Apache OpenWhisk or IBM Cloud Functions"
 
-## Hello World Kotlin Action
-
-The runtime for Kotlin supports two types of API for creating actions:  
-1. JSON parameter and return type, using the GSON Library.  
-2. Data Class parameter and return type.
-
-### JSON parameter based main.kt
-The following shows how to build a "HelloWorld" action using JSON parameters:
-
-```kotlin
-import com.google.gson.JsonObject
-
-fun main(args: JsonObject) : JsonObject {
-    val name = args.getAsJsonPrimitive("name").getAsString();
-    val hello = JsonObject();
-    hello.addProperty("greeting", "Hello " + name + "!");
-    return hello
-}
-```
-The action can then be compiled into a JAR file for use with the runtime for Kotlin. As the action has a dependency on the [Google GSON library](https://github.com/google/gson), it needs to be available on the classpath:
 ```sh
-kotlinc -classpath ./gson-2.6.2.jar main.kt -d myAction.jar
+bx wsk action update myAction myAction.jar --docker ibmfunctions/action-kotlin
 ```
 
```

Comment:
We should explain how the user would specify the full qualified Class name if putting Class ina package and also how to specify the method if not main, and if .kt file is not main the the name of the file would be use as Class name but this would need to specify with “—main” in the CLI

Reply:
sweet 👍 Thanks @seabaylea